extensions/panache/hibernate-orm-rest-data-panache/deployment/src/main/java/io/quarkus/hibernate/orm/rest/data/panache/deployment/HibernateOrmPanacheRestProcessor.java

```diff
@@ -134,6 +154,14 @@ private void validateResource(IndexView index, ClassInfo classInfo) {
         }
     }
 
+    private List<ClassInfo> getListenersByEntityType(List<ResourceMethodListenerBuildItem> resourceMethodListeners,
+            String entityTypeName) {
+        return resourceMethodListeners.stream()
+                .filter(e -> e.getEntityType().name().toString().equals(entityTypeName))
```

Comment:
Just a heads-up: this doesn't seem to handle inheritance.

Reply:
Thanks for spotting! I've already addressed this.